### PR TITLE
Fetch content after another content item

### PR DESCRIPTION
**Pull request by alecgibson**
This change adds the ability to fetch the content item after a given
content item, according to the sort criteria provided in a query.

This is particularly useful when trying to pick the “next” item when
viewing a single content item.

The previous approach to this used the list of content items on a page,
but that approach breaks when viewing the last item on a page.

## Dependencies

- [x] https://github.com/alphagov/content-performance-manager/pull/251

### CODE REVIEW

**Comment by alecgibson** [comment]
[If you click the last item on a list page, you don't get a 'next' link](https://trello.com/c/B8YQWsfv/308-if-you-click-the-last-item-on-a-list-page-you-dont-get-a-next-link)

File: spec/domain/content/query_spec.rb [hunk 1]
```diff
@@ -143,8 +144,31 @@
...
+        expect(subject.content_items).to contain_exactly(content_item_3)
+      end
+    end
+
+    describe "sorting by page views ascending" do
+      before do
+        subject.sort_direction(:asc)
+      end
+
+      it "can pick items after another item" do
```
Comment (pmanrubia) [inline spec/domain/content/query_spec.rb]: Could you please isolate in a `describe` the tests for `after`.
It would be good to test what happens when the element is the last position, last position of a page, first position, ..., in the middle. We have had so many issues with this feature, that it would be good to describe all the edge cases.

Comment (pmanrubia) [inline spec/domain/content/query_spec.rb]: Could you please add an integration test where you replicate the pagination issue?

Comment (alecgibson) [inline spec/domain/content/query_spec.rb]: > It would be good to test what happens when the element is [in the] ... last position of a page

What do you mean? This `after` function has literally zero idea about paging.

Comment (pmanrubia) [inline spec/domain/content/query_spec.rb]: Usually, you look for the next item within a list. And it is in this context where we have had all the issues. For example, would it return nil if there is no `content_item` after the current one?

You can have integration tests that exercise this query in different contexts, and simplify unit tests?

**Comment by alecgibson** [comment]
@pmanrubia I've added some more tests.

**Comment by alecgibson** [comment]
[When filtering by non-audited items, saving an audit won't take you to the next item to audit.](https://trello.com/c/9gtTPirg/326-when-filtering-by-non-audited-items-saving-an-audit-wont-take-you-to-the-next-item-to-audit)
